libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -662,6 +663,74 @@ class MapboxNavigation(
         )
     }
 
+    /**
+     * Create and cache a user feedback about an issue or problem with the Navigation SDK.
+     *
+     * Instead of sending the feedback, this API will cache the feedback in the Navigation SDK. The cached feedbacks can be retrieved by calling [getCachedUserFeedback] so the user can update and send them in a later time. To send the cached feedbacks, please use [postCachedUserFeedback].
+     *
+     * If you want to send the feedback directly without caching it, you should use [postUserFeedback].
+     *
+     * @param feedbackType one of [FeedbackEvent.Type]
+     * @param description description message
+     * @param feedbackSource one of [FeedbackEvent.Source]
+     * @param screenshot encoded screenshot (optional)
+     * @param feedbackSubType array of [FeedbackEvent.Description] (optional)
+     * @param appMetadata [AppMetadata] information (optional)
+     *
+     * @see [postUserFeedback]
+     * @see [getCachedUserFeedback]
+     * @see [postCachedUserFeedback]
+     */
+    fun cacheUserFeedback(
+        @FeedbackEvent.Type feedbackType: String,
+        description: String,
+        @FeedbackEvent.Source feedbackSource: String,
+        screenshot: String?,
+        feedbackSubType: Array<String> = emptyArray(),
+        appMetadata: AppMetadata? = null
+    ) {
+        MapboxNavigationTelemetry.cacheUserFeedback(
+            feedbackType,
+            description,
+            feedbackSource,
+            screenshot,
+            feedbackSubType,
+            appMetadata
+        )
+    }
+
+    /**
+     * Get a list of [CachedNavigationFeedbackEvent] which was created by [cacheUserFeedback].
+     *
+     * You can update the cached feedback to have more information like [CachedNavigationFeedbackEvent.feedbackSubType] and [CachedNavigationFeedbackEvent.description] or other fields. And use [postCachedUserFeedback] to send the updated cached feedbacks.
+     *
+     * @see [cacheUserFeedback]
+     * @see [postCachedUserFeedback]
+     *
+     * @return a list of [CachedNavigationFeedbackEvent]s
+     */
+    fun getCachedUserFeedback() = MapboxNavigationTelemetry.getCachedUserFeedback()
```

Comment:
looks like we have the next flow
- cache events with `MapboxNavigationTelemetry`
- take them and add some details (what class is responsible for such changes?)
- use `MapboxNavigationTelemetry` to post events

is it a good idea to take events out of `MapboxNavigationTelemetry` to change them?
what happens if user closes the app/ cancel the session? with current implementation all pending events will be sent immediately. But looks like cached events will be lost

Reply:
that's correct. 
* the flow you described is correct. That's why you can see I expose a new `CachedNavigationFeedbackEvent` instead of the internal `NavigationFeedbackEvent` and also only allow the `feedbackSubType` and `description` to be modified.
* yes, we take the risk of losing those cached events base on agreement of the feedback 2.0 flow discussion. I mentioned you in that ticket for reference.